### ISSUE TITLE
Brand payload packets don't work

### ISSUE THREAD

**Issue by ghost** [PaperMC/Paper#9904]
### Expected behavior

Client brand packets should contain their brand.

### Observed/Actual behavior

After recently updating paper, it seems like somehow brand packets are deserialized wrongly to have unknown brand payloads instead of net.minecraft.network.protocol.common.custom.BrandPayload, the result is that the .data() is an empty string instead of the client brand.

### Steps/models to reproduce

Listen for payload packets

### Plugin and Datapack List

Protocollib (for listening to the packets), i am just calling .getHandle() to get the raw NMS packet though

### Paper version

This server is running Paper version git-Paper-"1865625" (MC: 1.20.2) (Implementing API version 1.20.2-R0.1-SNAPSHOT) (Git: 1865625)
You are 3 version(s) behind
Download the new version at: https://papermc.io/downloads/paper
Previous version: git-Paper-"4b0bc74" (MC: 1.20.2)

**Comment by Machine-Maker** [comment]
Yeah, [upstream](https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/commits/2a381856b95784163eaca2213707bb48e1e2003e#nms-patches/net/minecraft/network/protocol/common/ServerboundCustomPayloadPacket.patch) just decided to remove the custom handling and handle everything as unknown.

**Comment by Sytm** [comment]
There is #9863 as a possible approach to solve this issue
